### ISSUE TITLE
[Pagination]: improve order of select labeling for screen readers

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon#16538]
## Acceptance criteria

From Michael Gower as part of omnibus website doc issue [#3238](https://github.com/carbon-design-system/carbon-website/issues/3238)

### Pagination a11y issues

**Labeling**
The labeling is not great for a screen reader, and there could be ways of making it better. The simplest is changing the design spec to be:
- "Page: [12 of 40] v" instead of "[12] v of 40 pages"
- I've used the "[ ] v" to represent where the input and down chevron is/could be. That would need to be tested, but with zero real difference to a sighted user could likely be a much better screen reader or magnified result. Note that this also works when the unknown variant is enabled

![image](https://user-images.githubusercontent.com/14298245/199504829-60e5dc30-55b1-49f9-b459-6c0234776c38.png)

![image](https://user-images.githubusercontent.com/14298245/199504860-b0bc93da-02d3-4af6-9822-d337d3932d70.png)

- Speaking of putting "Page" in front of the right-side input, almost all the variations in Figma already show that, so it would result in a more consistent layout

<img width="366" alt="image" src="https://user-images.githubusercontent.com/14298245/199510541-e5733f21-296c-45e7-b176-f71b4132a09c.png">

----

**Other dev improvements to consider**
There are more holistic changes that could be made, but this would be a pretty easy one to do.
- The current design also requires you to do something with the 'orphan' text when it comes to building a label. At the moment, Carbon has used a hidden aria label that is populated based on values on screen, but I think an `aria-describedby` is more elegant and also better supports internationalization.

`<label for="cds-pagination-select-id-5-right" class="cds--label cdes--visually hidden">Page number, of 6 pages</label>
`
- They could just take the select itself and use aria-describedby to link it with an ID they assign to

`<span class"cds--pagination__text">of 6 pages</span>`

- The speech output with Jaws is much smoother, IMO.
- I've made a small recording of [current](https://ibm.box.com/s/tg6h65u1zixebmp5ilc9fz3rh5a6w6gs) and [improved](https://ibm.box.com/s/zdfg0x5hlb0tu9uc2cybgv4tm46x2fwc) screen reader behavior. If combined with the above suggestion on labeling changes, this would result in real improvement.

- The chunk of text after the first input ("1 to 10 of 103 items" in the image above) should be tested as a live region. The advantages of this: when a user incremented the prior or next pages, or when a user changed the number of records shown OR potentially when the filtering was altered, this would be announced by the screen reader. Could be VERY useful.

**Comment by laurenmrice** [comment]
Here are my thoughts on this one. The only suggestion I would make that is different than Gower's is to make sure that the chevron still clearly belongs to the changing value (12), which matches user expectation for a select.

<img width="1056" height="1360" alt="Image" src="https://github.com/user-attachments/assets/a2d241ac-b7c5-450e-81df-f5abf2886944" />
